src/utils.ts

```diff
@@ -76,11 +77,19 @@ export function validateDataSourcePropertyName(name: string): string | undefined
 	}
 }
 
+export function validateResourceName(name: string): string | undefined {
+	if (name && /^[a-z]{1}[a-z0-9.\-]{3,252}$/.test(name)) {
```

Comment:
the regexp is still not accurate.
I have this message:

`Invalid value: "asecret-1-": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`

at least now we have the real regexp used that we can replace :-)

Reply:
I am rather unsure if that regexp is correct. it seems to check for a domain name.

Reply:
if the regexp that I mentioned is incorrect, you need to report an issue to Kubernetes or kubectl

Reply:
ok, maybe. Then the erro message in kubectl shoudl be improved